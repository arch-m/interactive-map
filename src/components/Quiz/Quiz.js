import React, { useState } from 'react'
import getRandomColor from '../Map/Colors'
import Style from './QuizScreen.module.css'

export default function Quiz({ questions }) {

  const [currentQuestion, setCurrentQuestion] = useState(0);
  const [showScore, setShowScore] = useState(false);
  const [score, setScore] = useState(0);
  const backgroundStyle = {
    background: `linear-gradient(167deg, ${getRandomColor("pasteles")} 0%, ${getRandomColor("solidos")} 100%)`,
  }

  const handleAnswerOptionClick = (isCorrect) => {
    if (isCorrect) {
      setScore(score + 1);
    }

    const nextQuestion = currentQuestion + 1;
    if (nextQuestion < questions.length) {
      setCurrentQuestion(nextQuestion);
    } else {
      setShowScore(true);
    }
  };
  return (
    <div style={backgroundStyle} className='h-100 d-flex flex-column justify-content-center align-items-center' >
      {showScore ? (
        <h1 className={Style.header}>
          Resolviste bien {score} de {questions.length}!
        </h1>
      ) : (
          <>
          <div>
            <h1 className={Style.header}>
              Pregunta {currentQuestion + 1}/{questions.length}
            </h1>
            <h2 className={Style.question}>
              {questions[currentQuestion].questionText}
            </h2>
          </div>
          <div className={`row row-cols-2 ${Style.answers}`}>
              {questions[currentQuestion].answerOptions.map((answerOption) =>
                <div className="p-0 py-3 px-3 col">
                  <button key={answerOption.answerText} className={`btn ${Style.btnResp}`} onClick={() => handleAnswerOptionClick(answerOption.isCorrect)}>{answerOption.answerText}</button>
                </div>
              )}
            </div>
          </>
      )}
    </div>
  );

}