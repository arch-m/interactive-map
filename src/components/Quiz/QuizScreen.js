import React from 'react';

export const QuizScreen = () => {
  return (
    <div>
      quiz
    </div>
  );
}
