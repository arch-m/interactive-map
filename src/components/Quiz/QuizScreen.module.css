.header, .question{
    color: white;
    text-align: center;
    text-shadow: 1px 1px 4px rgba(0,0,0,0.5);
    font-weight: 700;
    animation: fadeIn 2s;
    opacity: 1;
}

.header{
    font-size: 3rem;
}

.question{
    font-size: 2rem;
}

.btnResp{
    width: 100%;
    font-size: 1.2rem;
    font-weight: 600;
    animation: fadeIn 2s;
    color: #5a5a5a;
    background-color: white;
    height: 5rem;
    -webkit-box-shadow: 3px 3px 9px 0px rgba(162,162,162,0.68); 
    box-shadow: 3px 3px 9px 0px rgba(162,162,162,0.68);
}

.btnResp:hover{
    background-color: #4c98bb;
    color: white;
}


@keyframes fadeIn {
    0% {opacity: 0;}
    100% {opacity: 1;}
}

@-moz-keyframes fadeIn {
    0% {opacity: 0;}
    100% {opacity: 1;}
}

@-webkit-keyframes fadeIn {
    0% {opacity: 0;}
    100% {opacity: 1;}
}

@-o-keyframes fadeIn {
    0% {opacity: 0;}
    100% {opacity: 1;}
}

@-ms-keyframes fadeIn {
    0% {opacity: 0;}
    100% {opacity: 1;}
}