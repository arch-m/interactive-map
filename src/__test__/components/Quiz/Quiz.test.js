import { shallow } from "enzyme";
import Quiz from "../../../components/Quiz/Quiz";

describe('Pruebas a Quiz', () => {

  const handleAnswerOptionClick = jest.fn();

  const questions = [
    {
      questionText: 'Pregunta 1',
      answerOptions: [
        {answerText: 'respuesta 1', isCorrect: false },
        {answerText: 'respuesta 2', isCorrect: true },
        {answerText: 'respuesta 3', isCorrect: false },
        {answerText: 'respuesta 4', isCorrect: false },
      ]
    },
    {
      questionText: 'Pregunta 2',
      answerOptions: [
        {answerText: 'respuesta 1', isCorrect: false },
        {answerText: 'respuesta 2', isCorrect: true },
        {answerText: 'respuesta 3', isCorrect: false },
        {answerText: 'respuesta 4', isCorrect: false },
      ]
    },
  ];
  const wrapper = shallow(<Quiz questions={ questions } />);

  test('Debe mostrar el componente correctamente', () => {
    expect ( wrapper ).toMatchSnapshot();
  });

  test('Debe llamar cambiar el estado de la pregunta actual cuando se selecciona una respuesta', () => {

    wrapper.find('button').at(1).prop('onClick')();

    // expect( setCurrentQuestion ).toHaveBeenCalled();
    expect( handleAnswerOptionClick ).toHaveBeenCalled();

  })
  

})
