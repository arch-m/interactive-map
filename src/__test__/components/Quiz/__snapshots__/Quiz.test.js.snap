// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a Quiz Debe mostrar el componente correctamente 1`] = `
<div
  className="app"
>
  <div
    className="question-section"
  >
    <div
      className="question-count"
    >
      <span>
        Question 
        1
      </span>
      /
      2
    </div>
    <div
      className="question-text"
    >
      Pregunta 1
    </div>
  </div>
  <div
    className="answer-section"
  >
    <button
      key="respuesta 1"
      onClick={[Function]}
    >
      respuesta 1
    </button>
    <button
      key="respuesta 2"
      onClick={[Function]}
    >
      respuesta 2
    </button>
    <button
      key="respuesta 3"
      onClick={[Function]}
    >
      respuesta 3
    </button>
    <button
      key="respuesta 4"
      onClick={[Function]}
    >
      respuesta 4
    </button>
  </div>
</div>
`;
